src/components/search/Filter/Filter.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useState } from 'react'
+import type { KeyboardEvent, MouseEvent } from 'react'
+import { useSearch } from '@faststore/sdk'
+import type {
+  IStoreSelectedFacet,
+  FacetedFilter_FacetsFragment,
+} from '@generated/graphql'
+import {
+  Icon as UIIcon,
+  List as UIList,
+  Modal as UIModal,
+  Accordion as UIAccordion,
+  AccordionItem as UIAccordionItem,
+  AccordionButton as UIAccordionButton,
+  AccordionPanel as UIAccordionPanel,
+} from '@faststore/ui'
+import useWindowDimensions from 'src/hooks/useWindowDimensions'
+import Button from 'src/components/ui/Button'
+import Checkbox from 'src/components/ui/Checkbox'
+
+interface Props {
+  facets: FacetedFilter_FacetsFragment[]
+  /*
+   * Control whether the filter modal is open. (mobile only)
+   */
+  isOpen?: boolean
+  /**
+   * This function is called whenever the user hits "Escape", clicks outside
+   * the filter modal or clicks in close button. (mobile only)
+   */
+  onDismiss?: (event: MouseEvent | KeyboardEvent) => void
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+}
+
+function Filter({
+  facets,
+  onDismiss,
+  isOpen = false,
+  testId = 'store-filter',
+}: Props) {
+  const { width: screenWidth } = useWindowDimensions()
+  const { toggleFacet, toggleFacets, state: searchState } = useSearch()
+
+  const [expandedIndices, setExpandedIndices] = useState<Set<number>>(
+    new Set([])
+  )
+
+  const [selectedFilters, setSelectedFilters] = useState<IStoreSelectedFacet[]>(
+    searchState.selectedFacets ?? []
+  )
+
+  const [isMobile, setIsMobile] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (screenWidth) {
+      // notebook breakpoint = 1280px (See breakpoints on styles/global.scss)
+      setIsMobile(screenWidth < 1280)
+    }
+  }, [screenWidth])
+
+  const onAccordionChange = (index: number) => {
+    if (expandedIndices.has(index)) {
+      expandedIndices.delete(index)
+      setExpandedIndices(new Set(expandedIndices))
+
+      return
+    }
+
+    setExpandedIndices(new Set(expandedIndices.add(index)))
+  }
+
+  const onFilterChange = (item: IStoreSelectedFacet) => {
+    if (selectedFilters.some((filter) => filter.value === item.value)) {
+      const indexToRemove = selectedFilters.findIndex(
+        (f) => f.value === item.value
+      )
+
+      selectedFilters.splice(indexToRemove, 1)
+      setSelectedFilters([...selectedFilters])
+
+      return
+    }
+
+    setSelectedFilters([...selectedFilters, item])
+  }
```

Comment:
Not sure about that but since we depend on previous values should we refactor [to use the second form of setState()](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous) that accepts a function rather than an object on the 2 functions above?